main.go

```diff
@@ -88,9 +94,34 @@ func handler(w http.ResponseWriter, r *http.Request) {
 	log.Debugf("Scrape of target '%s' with module '%s' took %f seconds", target, moduleName, duration)
 }
 
+func updateConfiguration(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "POST":
+		rc := make(chan error)
+		reloadCh <- rc
+		if err := <-rc; err != nil {
+			http.Error(w, fmt.Sprintf("failed to reload config: %s", err), http.StatusInternalServerError)
+		}
+	default:
+		log.Errorf("POST method expected")
+		http.Error(w, "POST method expected", 400)
+	}
+}
+
+func (sc *SafeConfig) reloadConfig(configFile string) (err error) {
+	sc.Lock()
+	sc.C, err = config.LoadFile(configFile)
```

Comment:
If there's an error in the config this sill updates the config. We should only update `sc.C` after checking `err`.